pilot/cmd/pilot-agent/main_test.go

```diff
@@ -164,3 +165,44 @@ func TestPilotSanIfAuthenticationMutualStdDomainConsul(t *testing.T) {
 
 	g.Expect(pilotSAN).To(gomega.Equal([]string{"spiffe:///ns/anything/sa/istio-pilot-service-account"}))
 }
+
+func TestGetDomainName(t *testing.T) {
+	tests := []struct {
+		name       string
+		resolvConf string
+		want       string
+	}{
+		{
+			name: "all good",
+			resolvConf: `
+nameserver 1.1.1.1
+search default.svc.abc.com svc.abc.com abc.com
```

Comment:
are you sure this will work? for the global services in multicluster, we are adding additional search domains (.global, etc.)..

Reply:
please provide me with a sample of /etc/resovl.conf you are talking about here, btw this resolv.conf is the one which automatically mounted by kubelet into the pod.

Reply:
@sbezverk there is a flag that sets custom dnsConfig name searches in *EVERY* sidecar.  We absolutely don't want to break that functionality - it took me 3-4 weeks to get working after someone else had spent ages laying the groundwork.  Take a look at https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/values-istio-multicluster-gateways.yaml#L3-L5.  Probably making your life more difficult, the search namespace changes depending on the pod namespace.

Reply:
I think this is pretty cool - a test case for resolve.conf.  the dnsConfig overrides don't get added unless a special map is set in the yaml.

Reply:
@sdake @rshriram  I do not propose to get rid of a domain flag, but it seems it does not cover all possibilities, example incorrectly rendered mixer templates by pilot-agent, another thing this flag is not mandatory so use with a non-default cluster name easily might miss it resulting a non-working application. I suggest to use this func as a safety feature, if `domain` is empty, it gets called and it will populate the actual cluster's domain name. wdyt?